wandb/sdk/lib/apikey.py

```diff
@@ -179,20 +217,32 @@
                 key_suffix, len(key_suffix)
             )
         )
-        return None
-    try:
-        normalized_host = urlparse(host).netloc.split(":")[0]
-        netrc_path = get_netrc_file_path()
-        wandb.termlog(
-            f"Appending key for {normalized_host} to your netrc file: {netrc_path}"
+        return False
+
+    normalized_host = urlparse(host).netloc
+    netrc_path = get_netrc_file_path()
+    netrc_access = check_netrc_access(netrc_path)
+
+    if not netrc_access.write_access or not netrc_access.read_access:
+        wandb.termwarn(
+            f"Cannot access {netrc_path}. In order to persist your API key,"
+            + "\nGrant read & write permissions for your user to the file,"
+            + '\nor specify a different file with the environment variable "NETRC={new_netrc_path}".'
         )
-        machine_line = f"machine {normalized_host}"
-        orig_lines = None
-        try:
-            with open(netrc_path) as f:
-                orig_lines = f.read().strip().split("\n")
-        except OSError:
-            pass
+        return False
+
+    machine_line = f"machine {normalized_host}"
+    orig_lines = None
+    try:
+        with open(netrc_path) as f:
+            orig_lines = f.read().strip().split("\n")
+    except FileNotFoundError:
+        wandb.termlog("No netrc file found, creating one.")
+    except OSError:
+        wandb.termerror(f"Unable to read {netrc_path}")
```

Comment:
nit: this discards information about the error that could be useful for debugging, but I'm not sure we can use a logger here.

See my other comment about potentially letting errors bubble. I would rewrite this `except` like so:

```py
except OSError as e:
    raise WriteNetrcError(f"Unable to read {netrc_path}") from e
```